### PR TITLE
 [videoplayer] Update selectionstream's properties that can change over time in UpdateStreamInfos().

### PR DESCRIPTION
**Pull request by ace20022**
…gt if codec->width/height is zero.

### CODE REVIEW

**Comment by ace20022** [comment]
@FernetMenta "Fixes" osd for http://demo-uhd3d.com/fiche.php?cat=uhd&id=101. The correct height is detected later at read(), but videplayer's SelectionStream vector is not updated.
My vision is to add a callback to fill/update SelectionStreams rather than adding them once at opendemuxer. But I have some other stuff in the pipeline first.

**Comment by FernetMenta** [comment]
we have CVideoPlayer::UpdateStreamInfos() to update parameters that can change or only can be detected by steam players. does this not fit here?

**Comment by ace20022** [comment]
> we have CVideoPlayer::UpdateStreamInfos() to update parameters that can change or only can be detected by steam players. does this not fit here?

Yes that works if we query the demuxer in that method, the videoplayervideo doesn't know the dimension if I'm not wrong. Is this the way you prefer?

**Comment by FernetMenta** [comment]
@ace20022 I am fine with this pr too. Just looked into this and noticed that we already update SrcRect and DstRect there. You know that coded_width can slightly differ from width, don't you?

**Comment by ace20022** [comment]
> You know that coded_width can slightly differ from width, don't you?

Yes, with that file it's more than slightly 3840 vs. 0 ;) 

Is SrcRect reliable?

**Comment by FernetMenta** [comment]
> Yes, with that file it's more than slightly 3840 vs. 0 ;)

hehe, I was more referring to 1088 vs 1080

> Is SrcRect reliable?

in full screen it is the same as width/height but may be clipped when used by video control. CVideoPlayer::UpdateStreamInfos is called by the videoplayer thread. you can safely update from demuxer data too.

**Comment by ace20022** [comment]
Yup, must have missed that method. Good I know it now, that solves (hopefully) another codec detection for dvds ;)

**Comment by ace20022** [comment]
@FernetMenta updated and added a commit for dvds. Could you please double-check that I used the correct stream id.

File: xbmc/cores/VideoPlayer/VideoPlayer.cpp [hunk 1]
```diff
@@ -3227,6 +3227,13 @@ void CVideoPlayer::UpdateStreamInfos()
     s.stereo_mode = m_VideoPlayerVideo->GetStereoMode();
     if (s.stereo_mode == "mono")
       s.stereo_mode = "";
+
+    CDemuxStream* stream = m_pDemuxer->GetStreamFromVideoId(streamId);
+    if (stream && stream->type == STREAM_VIDEO)
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoPlayer.cpp]: I would rather use m_pDemuxer->GetStream(m_CurrentVideo.id)
GetStreamFromVideoId is used nowhere in the code and should be removed.

**Comment by ace20022** [comment]
@FernetMenta changed both, okay?

jenkins build this please

**Comment by FernetMenta** [comment]
+1

**Comment by ace20022** [comment]
jenkins build this please

**Comment by ace20022** [comment]
One more time...

jenkins build this please

**Pull request merged**